library/packages/src/lib/y2packager/licenses_fetchers.rb

```diff
@@ -11,21 +11,18 @@
 # ------------------------------------------------------------------------------
 
 require "yast"
+require "y2packager/licenses_fetchers/rpm"
+require "y2packager/licenses_fetchers/url"
+require "y2packager/licenses_fetchers/dummy"
 
 module Y2Packager
-  # This class is responsible for storing the different license translations.
-  class LicenseTranslation
-    # @return [String] Content Language
-    attr_reader :lang
-    attr_reader :content
-
-    # Constructor
+  module LicensesFetchers
+    # Return the licenses proper fetcher for a given source
     #
-    # @param language [String]
-    # @param content [String] of the license for the given language
-    def initialize(language:, content:)
-      @lang = language
-      @content = content
+    # @param source [Symbol] :rpm or :url
+    def self.for(source, product_name, options = {})
```

Comment:
As I mention above I think nicer API would be detect automatically source. Just try to find default lang for all known source and what finds first, then use such fetcher.

Reply:
Sure, that API would be nicer. But at this point, we were requested to keep changes under control and, in `registration` and `packager` modules, the license content is obtained using different mechanisms.

In the future, that code should be moved to its own fetchers/handlers and, using proper objects, we could autodetect the right source and adapt to it accordingly.

We will write some documentation about what's expected in the future. For instance, a method like `ProductLicense#find_or_create` should not exist, but it's there just to make the transition from the old code.